src/store.cc

```diff
@@ -393,9 +393,6 @@ destroyStoreEntry(void *data)
     StoreEntry *e = static_cast<StoreEntry *>(static_cast<hash_link *>(data));
     assert(e != NULL);
 
-    if (e == NullStoreEntry::getInstance())
```

Comment:
This function is used by hashFreeItems during Controller destruct (shutdown?) without validation that the data passed in is non-nil. This if-statement implies that at somepoint 'e' was found to be NullStoreEntry.

Has this case been tested and found working? if not it would seem prudent to leave this as if(!e) and remove the assert above.

Reply:
The old code was needed to assure that the NullStoreEntry _singleton_ (accidentally added to store_table or whatever) is never deleted. Patched code does not have NullStoreEntry objects at all so the old check is both pointless and impossible.

The old assert is correct (if paranoid). We should not remove/relax it. Nil store_table entries are still prohibited and, AFAICT, all other existing destroyStoreEntry() callers are OK.